jihyeon/services/userService.js

```diff
@@ -1,22 +1,41 @@
 const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
 
-const createUserDao = require("../models/UserDao");
+const userDao = require("../models/userDao");
 const pwValidationCheck = require("../utils/validation-check.js");
 
+const checkHash = async (password, hashedPassword) => {
+  return await bcrypt.compare(password, hashedPassword);
+};
+
 const signUp = async (email, password, name, age, phoneNumber) => {
   await pwValidationCheck(password);
 
   const hashedPassword = await bcrypt.hash(password, 12);
+  return userDao.createUser(email, hashedPassword, name, age, phoneNumber);
+};
+
+const login = async (email, password) => {
+  const userInfo = await userDao.login(email);
+
+  const storedPassword = userInfo[0]["password"];
+  const userId = userInfo[0]["id"];
+
+  const checkPassword = await checkHash(password, storedPassword);
+
+  const payLoad = { user_id: userId };
+  const secretKey = process.env.SECRETKEY;
+
+  if (!checkPassword) {
+    const err = new Error("Invalid User");
+    err.statusCode = 409;
+    throw err;
+  }
```

Comment:
```suggestion
  const [userInfo] = await userDao.getUserByEmail(email);

  if (!user || !bcrypt.compare(password, user.password)) throw new Error("Invalid Email or Password");
 
  return jwt.sign({ userId: user.id }, process.env.SECRETKEY);
```
suggestion을 참고해서 리팩토링해주세요! 

또한 (payload, secretKey) 같이 가독성을 해치지않는 선에서 반복되어 사용되는 변수가 아니라면, 메모리 소비 없이 곧바로 필요한 부분에 작성될 수 있게 해주세요.